Source/WebCore/accessibility/AccessibilityObject.cpp

```diff
@@ -404,6 +404,39 @@ std::optional<SimpleRange> AccessibilityObject::misspellingRange(const SimpleRan
     return std::nullopt;
 }
 
+bool AccessibilityObject::isNodeForComposition(const Editor& editor) const
+{
+    auto* compositionNode = editor.compositionNode();
+    auto* cache = axObjectCache();
+    if (!compositionNode || !cache)
+        return false;
+
+    auto* compositionObject = cache->textCompositionObjectForNode(*compositionNode);
+    if (!compositionObject)
+        return false;
+
+    return compositionObject->objectID() == objectID();
+}
+
+std::optional<CharacterRange> AccessibilityObject::textInputMarkedRange() const
+{
+    auto* node = this->node();
```

Comment:
This should be a `WeakPtr node` too (sorry for missing this in the last round).

Reply:
cool, sounds good!